### DISCUSSION TITLE
Using spectral libraries for modified peptides + library form  FASTA

### DISCUSSION THREAD

**Discussion by Apirog9** [vdemichev/DiaNN#173] [Q&A]
If I want to use all modified peptides (e.g deamidated, pyro-glu) from a spectral library I understand that I should add modifications using --mod option before starting analysis. If I additionally turn on:
- FASTA digest for library-free search
- Deep learning based spectra, RT, IM prediction

Will setting the modifications affect analysis? Probably it would be preferable to leave spectra of modified precursors as is.

**Comment by vdemichev** [comment]
You don't need to add modifications, unless DIA-NN is instructed to generate a library from DIA data and library generation mode is set to 'smart profiling' or 'full profiling'.

Using FASTA digest together with a library does not make sense.
No point in using deep learning spectra prediction, if the spectra in the library are good. Makes sense only if you know that they are not good.

**Reply by Apirog9** [reply]
Yes, however I am speculating about the scenarios when library spectra are good but the library is far from complete(just several DDA runs, or runs of modification-enriched fraction). Then I would like to use both input library and library generated from DIA runs. In such case, only spectra from input library should include spectra of modified peptides.

**Reply by vdemichev** [reply]
Then you might want to create a library from DIA data in library-free mode, and then search the data again using both libraries, see also https://github.com/vdemichev/DiaNN/discussions/144. RT scale in the DDA library must match Biognosys iRT, or, alternatively, the DDA library need to be converted, before the final analysis, to a predicted library using deep learning option in DIA-NN.
